public/js/ticket.js

```diff
@@ -1,16 +1,67 @@
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const deleteTicket = id => {
+function deleteTicket(id) {
   if (confirm('Biztosan törlöd?')) {
     fetch(`/tickets/${id}`, { method: 'DELETE' })
-      .then(async (res) => {
-        if (res.status === 204) {
+      .then(async res => {
+        switch(res.status) {
+        case 204:
           const ticket = document.getElementById(`ticket-${id}`)
           ticket.parentNode.removeChild(ticket)
-        } else if (res.status === 404) {
+          break
+        case 404:
           data = await res.json()
           displayMessage('danger', data.message)
+          break
         }
       })
       .catch(err => displayMessage('danger', err))
   }
 }
+
+function validateTicket(data) {
+  const errors = []
+  const room = parseInt(data.get('roomNumber'))
+  const description = data.get('description')
+
+  if (typeof room !== 'number' || room < 3 || room > 18) {
+    errors.push('A szint csak 3 és 18 közötti szám lehet')
+  }
+  if (!description) {
+    errors.push('A leírás nem lehet üres')
+  }
+  if (description.length > 500) {
+    errors.push('A leírás max 500 karakter lehet')
+  }
+  return errors
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function addTicket() {
+  const formEl = document.getElementById('ticket-form')
+  const formData = new FormData(formEl)
+  const errors = validateTicket(formData)
+
+  if (errors.length) {
+    clearMessages()
+    errors.forEach(err => displayMessage('danger', err))
+  } else {
+    fetch('/tickets', { method: 'POST', body: formData })
+      .then(async res => {
+        switch (res.status) {
+        // The backend's render method doesn't really work when called
+        // from a fetch POST request, so this is the workaround for now
```

Comment:
If you use the`Content-type: application/json` HTTP header, then you can throw a 401 error from the `isAuthenticated` middleware.
```javascript
const contype = req.headers['content-type'];
if (contype?.indexOf('application/json') !== 0)
  return res.send(401);
```